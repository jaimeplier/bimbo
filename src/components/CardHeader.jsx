import React, { Component } from 'react';

export default class CardHeader extends Component {
  render() {
    return (
      <div className="row">
        <div className="col-6">
          <p className="card-title">{this.props.title}</p>
          <p className="small-text">{this.props.subtitle}</p>
        </div>
        <div className={`col-6 right f-right-i ${this.props.noMargin ? 'no-margin-i' : ''}`}>
          {this.props.rightContent}
        </div>
      </div>
    );
  }
}
