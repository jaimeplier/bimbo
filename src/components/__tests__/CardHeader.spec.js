import React from 'react';
import { shallow } from 'enzyme';
import CardHeader from '../CardHeader';

describe('CardHeader', () => {

  test('Component should render', () => {
    const cardHeader = shallow(<CardHeader />);
    expect(cardHeader).toMatchSnapshot();
  })

  test('Component should render title and subtext', () => {
    const titleText = "Title", subtitleText = "Subtitle";
    const cardHeader = shallow(<CardHeader title={titleText} subtitle={subtitleText} />)

    const title = cardHeader.find('.card-title');
    const subTitle = cardHeader.find('.small-text');

    expect(title.text()).toBe(titleText);
    expect(subTitle.text()).toBe(subtitleText);
  })

})
