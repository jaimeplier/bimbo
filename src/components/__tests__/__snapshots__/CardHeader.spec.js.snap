exports[`CardHeader Component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-6">
          <p
                    className="card-title" />
          <p
                    className="small-text" />
</div>,
        <div
          className="col-6 right f-right-i " />,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-title" />,
            <p
              className="small-text" />,
          ],
          "className": "col-6",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-title",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "small-text",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "col-6 right f-right-i ",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-6">
            <p
                        className="card-title" />
            <p
                        className="small-text" />
</div>,
          <div
            className="col-6 right f-right-i " />,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-title" />,
              <p
                className="small-text" />,
            ],
            "className": "col-6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card-title",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "small-text",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "col-6 right f-right-i ",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
